molecule/verifier/ansible.py

```diff
@@ -0,0 +1,95 @@
+#  Copyright (c) 2019 Red Hat, Inc.
+#
+#  Permission is hereby granted, free of charge, to any person obtaining a copy
+#  of this software and associated documentation files (the "Software"), to
+#  deal in the Software without restriction, including without limitation the
+#  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+#  sell copies of the Software, and to permit persons to whom the Software is
+#  furnished to do so, subject to the following conditions:
+#
+#  The above copyright notice and this permission notice shall be included in
+#  all copies or substantial portions of the Software.
+#
+#  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+#  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+#  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+#  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+#  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+#  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+#  DEALINGS IN THE SOFTWARE.
+
+import os
+
+from molecule import logger
+from molecule import util
+from molecule.verifier import base
+
+log = logger.get_logger(__name__)
+
+
+class Ansible(base.Base):
+    """
+    `Ansible`_ is not the default test runner.
+
+    Molecule executes a playbook (`verify.yml`) located in the role's
+    `scenario.directory`.
+
+    .. code-block:: yaml
+
+        verifier:
+          name: ansible
+          lint:
+            name: yamllint
+
+    The testing can be disabled by setting ``enabled`` to False.
+
+    .. code-block:: yaml
+
+        verifier:
+          name: ansible
+          enabled: False
+
+    Environment variables can be passed to the verifier.
+
+    .. code-block:: yaml
+
+        verifier:
+          name: ansible
+          env:
+            FOO: bar
+    """
+
+    def __init__(self, config):
+        """
+        Sets up the requirements to execute ``ansible`` and returns None.
+
+        :param config: An instance of a Molecule config.
+        :return: None
+        """
+        super(Ansible, self).__init__(config)
+
+    @property
+    def name(self):
+        return 'ansible'
+
+    @property
+    def default_options(self):
+        return {}
+
+    @property
+    def default_env(self):
+        return util.merge_dicts(os.environ.copy(), self._config.env)
```

Comment:
This overrides anything that was in env, effectively making setting any of those env vars useless. Is that what you really intended?

Reply:
I'm not sure, I did this because it's what the other verifiers were doing. Thinking about it more, I'll also merge the provisioner env, so we get the envvars set on the ansible provisioner.